docs/evm/cadence/batched-evm-transactions.md

```diff
@@ -0,0 +1,641 @@
+  (https://www.evm.flowscan.io/) explorers
+- Write code that interacts with the EVM via a COA
+
+## Prerequisites
+
+Before you dive in, make sure you have the following configured:
+
+- [MetaMask](https://metamask.io/download/) installed in your browser with an active account
+- [Flow Wallet extension](https://wallet.flow.com/download) installed in your browser with an active account
+- Both wallets funded with Testnet FLOW. See the [Faucet guide](../../ecosystem/faucets.md) for more information.
+
+## Overview
+
+For the purposes of demonstration, this walkthrough will focus on relatively simple EVM operations in addition to first
+creating a [Cadence-controlled EVM account (COA)](./interacting-with-coa.md). Specifically, we will:
+
+- Wrap FLOW as WFLOW
+- Approve an ERC721 to transfer WFLOW in exchange for an NFT mint
+- Mint an ERC721 token - this ERC721 has a 50% chance of failing (using [onchain VRF](../guides/vrf.md) to determine
+  success)
+ 
+These operations let us focus on the **core concepts** of this guide:
+
+1. **Batching EVM transactions** using Cadence
+2. **Conditioning execution** on the results of those EVM transactions.
+
+However, using these same principles, you'll have the power to address much more complex and interesting. For instance,
+replace wrapping FLOW with a DEX swap. Or instead of minting an ERC721, purchase an NFT listing from a marketplace.
+Combine these two and suddenly you can purchase NFTs with any ERC20 token, all in a single Cadence transaction,
+reverting everything if a single step fails.
+
+The point is, while a simple use case, this guide will give you the tools to build much more complex and interesting
+applications. So let's get started!
+
+## Components
+
+As mentioned in the [Overview](#overview), this guide involves three main actions:
+
+- Wrapping FLOW as WFLOW
+- Approving an ERC721 to transfer WFLOW in exchange for an NFT mint
+- Minting an ERC721 token
+
+Before interacting with these contracts, let's dig in a bit more into the components of this guide.
+
+### Wrap FLOW as WFLOW
+
+On Flow EVM, FLOW is the native currency and similar to other EVM platforms, the native currency is not accessible as an
+ERC20 token. To interact with ERC20 contracts, you need to wrap FLOW as WFLOW (Wrapped FLOW). This is Flow's equivalent
+of WETH on Ethereum.
+
+:::tip
+
+You can find WFLOW deployed to `0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e` on Flow
+[Testnet](https://evm-testnet.flowscan.io/token/0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e?tab=contract) &
+[Mainnet](https://evm.flowscan.io/token/0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e?tab=contract) and source code in the
+[`@onflow/flow-sol-utils` repository](https://github.com/onflow/flow-sol-utils).
+
+:::
+
+### Approve ERC721 Transfer
+
+Our example `MaybeMintERC721` contract accepts WFLOW in exchange for minting an NFT. However, the contract cannot move
```

Comment:
Great description of what wrapped Flow is, but _why_ would we need it here instead of just regular, unwrapped Flow?

Reply:
It's fine if it's just to demonstrate the complex chain of transactions, but I'd be explicit as to why.